test/unit/common/test_utils.py

```diff
@@ -4983,5 +4983,48 @@ def test_pairs(self):
                               (50, 60)]))
 
 
+class TestPivotTree(unittest.TestCase):
+    def _build_tree(self, *args):
+        tree = utils.PivotTree()
+        for node in args:
+            tree.add(node)
+        return tree
+
+    def test_bounds(self):
+        #   c
+        #  b d
+        #     e
+        tree = self._build_tree('c', 'b', 'd','e')
+        bounds = tree.get_pivot_bounds('e')
+        self.assertEqual(bounds, ('d', None))
+        bounds = tree.get_pivot_bounds('d')
+        self.assertEqual(bounds, ('c', None))
```

Comment:
The results of get_pivot_bounds() is a tuple of (>, <). That is the bounds is > the first result and < the last. When there is a None, that means there is no upper or lower bound. I think that was my plan anyway.

So with

   c
 b  d
       e
I'd expect the bounds of 'e' to be (d, None) or > d. What you have.

But the bounds of 'd' to be ('c', 'd') or > c and < d (because d has a child e that takes care > d.

But then what should we return if it's got 2 children, i.e not a leaf?